documentation/design/user-stories/shared-infra.adoc

```diff
@@ -182,17 +163,71 @@ metadata:
   name: endpoint1
   namespace: myapp
 spec:
-  # TLS configuration for endpoint
-  tls: {}
-
   protocols:
-    - AMQPS
-    - AMQP
+  - AMQP
+  - AMQPS
+  - AMQP-WS
+  - AMQP-WSS
 
+  # Optional: Override generated hostname for endpoint. Clients must have the AMQP OPEN hostname set to this value.
+  host: example.com
+
+  # Optional: specify TLS configuration for endpoint.
+  tls:
+    # Selfsigned is default unless specified by a `MessagingPlan`.
+    selfsigned: {}
+    # OpenShift requests a service certificate signed by OpenShift service CA
+    openshift: {}
+    # External allows providing certificate externally. CN/SANs are validated against host
+    external:
+      cert: # Reuse structure from address space connector spec
+      key: # Reuse structure from address space connector spec
+
+  # Cluster type creates a on-cluster load balancer
   cluster: {}
+
+  # NodePort type creates a nodePort service, exposing the application on an ephemeral port on all nodes.
+  nodePort: {}
+
+  # LoadBalancer type creates an external load balancer for this endpoint.
+  loadBalancer:
+    # Allow setting annotations on the service object that are specific to LoadBalancer implementations.
+    annotations:
+      aws-specific-annotation: value
+
+  # Ingress type creates an Ingress resource
+  ingress:
+    # Optional: Allow configuring annotations specific to ingress controller implementations
+    annotations: {}
+
+  # Route type maps to OpenShift Route resource
+  route:
+    # Allows controlling the TLS termination of this route. Only supported for AMQP-WSS protocol.
+    tlsTermination: passthrough | reencrypt
+
+status:
+  phase: Pending | Configuring | Active | Failed | Terminating
+  message: "Useful error message"
+
```

Comment:
should we add a type: like you have done for MessagingAddress?

Reply:
Agreed, good idea.